### PR TITLE
Make isort & black friends

### PR DESCRIPTION
**Pull request by mathfac**
## Motivation and Context
- Added `pre-commit` hooks for style checking
- Improved `isort` and `black` compatibility: added `toml/pyproject` support to isort pre-commit and added [an option recommended for compatibility](https://github.com/psf/black#how-black-wraps-lines)
- Added [seed-isort-config](https://github.com/asottile/seed-isort-config) pre-commit hook to manage third parties list
- Updated CI to use `pyproject` config for `isort` and `black`
- Moved `isort` config from `pyproject` to `.editorconfig`
- Added `ensure_newline_before_comments=true` to `isort` to be compatible with `black`. Will take action with next version of `isort`.
- Added habitat packages to `known_first_party`

## How Has This Been Tested
No conflicts with style checking.

## Types of changes

- Docs change / refactoring / dependency upgrade
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+known_third_party = ["PIL", "attr", "cv2", "gym", "habitat_sim", "imageio", "matplotlib", "numba", "numpy", "orbslam2", "pytest", "quaternion", "requests", "scipy", "setuptools", "torch", "torchvision", "tqdm", "version", "yacs"]
+multi_line_output = 3
+force_grid_wrap = false
+line_length = 79
+include_trailing_comma = true
+use_parentheses = true
+
+[tool.black]
+line_length = 79
+exclude = '''
```
Comment (apsdehal) [inline pyproject.toml]: Is there a possibility of picking this up from `.gitignore`?

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+[tool.isort]
```
Comment (apsdehal) [inline pyproject.toml]: There is a known_first_party option too.

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+known_third_party = ["PIL", "attr", "cv2", "gym", "habitat_sim", "imageio", "matplotlib", "numba", "numpy", "orbslam2", "pytest", "quaternion", "requests", "scipy", "setuptools", "torch", "torchvision", "tqdm", "version", "yacs"]
+multi_line_output = 3
```
Comment (apsdehal) [inline pyproject.toml]: Isort can also pick up .editorconfig setting, so maybe reuse that?

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
+exclude: 'build|src/deps|src/obsolete'
+
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v2.2.3
+    hooks:
+    -   id: trailing-whitespace
+    -   id: check-added-large-files
```
Comment (apsdehal) [inline .pre-commit-config.yaml]: Great!!

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+-   repo: https://github.com/asottile/seed-isort-config
+    rev: v1.9.2
+    hooks:
+    -   id: seed-isort-config
+
+-   repo: https://github.com/pre-commit/mirrors-isort
+    rev: v4.3.20
+    hooks:
+    -   id: isort
+        language_version: python3.6
```
Comment (erikwijmans) [inline .pre-commit-config.yaml]: I don't believe these are language versions are needed and they prevent you from using python3.7

File: pyproject.toml (hunk 1 above)
Comment (mathfac) [reply pyproject.toml]: black doesn't support [that feature yet](https://github.com/mdagaro/isort/commit/af012ef18e4680f2b3780d36bca9e3f25fa19a60), but it should run on files that are part of repo on `pre-commit` hook.

**Comment by mathfac** [comment]
Addressed the comments:
- Moved `isort` config from `pyproject` to `.editorconfig`
- Added `ensure_newline_before_comments=true` to `isort` to be compatible with `black`. Will take action with next version of `isort`.
- Added habitat packages to `known_first_party`

**Pull request merged**